packages/augur-core/source/contracts/reporting/AffiliateValidator.sol

```diff
@@ -20,16 +20,17 @@ contract AffiliateValidator is Ownable, IAffiliateValidator {
         operators[_operator] = false;
     }
 
-    function addKey(bytes32 _key, uint256 _salt, bytes32 _r, bytes32 _s, uint8 _v) external {
+    function addKey(bytes32 _key, address _account, uint256 _salt, bytes32 _r, bytes32 _s, uint8 _v) external {
         require(!usedSalts[_salt], "Salt already used");
-        bytes32 _hash = getKeyHash(_key, _salt);
+        bytes32 _hash = getKeyHash(_key, _account, _salt);
+        require(msg.sender == _account);
```

Comment:
Why not just pass `msg.sender` to `getKeyHash` here?

Reply:
The key could/will be for the signer who owns a gnosis safe which is making the tx

Reply:
hm actually it'll just have to be for the safe address anyway so yea, needs to just be msg.sender

Reply:
For context I was originally going to allow the safe to ad a key that was signed for the safe owner but that involved some major mess since the GnosisSafeRegistry is in Trading and not core.